### PR TITLE
Add a README for the API app

### PR DESCRIPTION
**Pull request by jchristgit**

### CODE REVIEW

File: pydis_site/apps/api/README.md [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+guide](https://www.django-rest-framework.org/tutorial/quickstart/) is a great
+resource to get started.
+
+## Directory structure
+
+Let's look over each of the subdirectories here:
+
+- `migrations` is the standard Django migrations folder. You usually won't need
+  to edit this manually, as `python manage.py makemigrations` handles this for
+  you in case you change our models.
```
Comment (janine9vn) [inline pydis_site/apps/api/README.md]: Would it be worth mentioning here our policy of renaming migrations to be human-readable/more informative? That might be diving too deep into our policy.

**Review by janine9vn — APPROVED** [review]
Fantastic and helps explain how our `api` folder is setup with wonderful resources as well. 

I have a small comment/note, but otherwise this makes it much more approachable with clear points/resources that people can run with and learn more about if they're feeling a bit lost.

**Review by MarkKoz — APPROVED** [review]
Maybe the migrations and models could use some more explanation (i.e. what those terms actually mean), but it could be covered elsewhere in a more general area I suppose.

File: pydis_site/apps/api/README.md (hunk 1 above)
Comment (jchristgit) [reply pydis_site/apps/api/README.md]: Yes!!!!!!!!!!!!

Comment (MarkKoz) [reply pydis_site/apps/api/README.md]: Is this the only app for which migrations are relevant? If not, we may want to have a more general documentation page to explain this, which ties into my review comment.

Comment (jchristgit) [reply pydis_site/apps/api/README.md]: I added this in 073e09a0. Thanks for the review!

**Comment by jchristgit** [comment]
> Maybe the migrations and models could use some more explanation (i.e. what those terms actually mean), but it could be covered elsewhere in a more general area I suppose.

For what it's worth, I don't think we should need to explain what these mean. We require basic Django knowledge, and if you've worked through the tutorial, you know what these mean. I don't want to duplicate Django documentation here either.

**Comment by jchristgit** [comment]
Thanks!

**Pull request merged**